vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -215,8 +221,16 @@ func run(healthCheck *metrics.HealthCheck, commonFlag *common.CommonFlags) {
 		postProcessors = append(postProcessors, &routines.IntegerCPUPostProcessor{})
 	}
 
+	var globalMaxAllowed apiv1.ResourceList
+	if !maxAllowedCPU.Quantity.IsZero() {
+		setGlobalMaxAllowed(&globalMaxAllowed, apiv1.ResourceCPU, maxAllowedCPU.Quantity)
+	}
+	if !maxAllowedMemory.Quantity.IsZero() {
+		setGlobalMaxAllowed(&globalMaxAllowed, apiv1.ResourceMemory, maxAllowedMemory.Quantity)
```

Comment:
Should there be some logic to ensure that the global max is greater than the global min?
It may be a bit strange since one is per container and the other per pod

Reply:
> However, I am not sure how to add such validation when the global min allowed flags are on Pod-level and global max allowed flags are on container-level.

Yeah, I agree. I can't figure out a way to make the validation work.

> In [#7147 (comment)](https://github.com/kubernetes/autoscaler/issues/7147#issuecomment-2501042262) I suggested to deprecate the global Pod-level min allowed flags because:

I don't think deprecation is necessary yet. Pod level resources [may be in the VPA's future](https://github.com/kubernetes/autoscaler/issues/7571), so those flags may be used for Pod level resources.